src/org/omegat/gui/exttrans/MachineTranslateTextArea.java

```diff
@@ -68,20 +78,31 @@
  * @author Jean-Christophe Helary
  * @author Aaron Madlon-Kay
  * @author Thomas Cordonnier
+ * @author Hiroshi Miura
  */
 @SuppressWarnings("serial")
 public class MachineTranslateTextArea extends EntryInfoThreadPane<MachineTranslationInfo>
         implements IPaneMenu {
 
     private static final String EXPLANATION = OStrings.getString("GUI_MACHINETRANSLATESWINDOW_explanation");
 
-    protected Set<MachineTranslationInfo> displayed = new HashSet<>();
+    /**
+     *  displayed hold entries.
+     *  <p>
+     *  It keeps the order by translatorName, MachineTranslationInfo is defined as Comparable&le;E&gt; that
+     *  prioritize translatorName. {@see MachineTranslationInfo}
+     *  A value should be unique and used for identify HTML element by id attribute.
+     *  We use an index counter for identifier of elements.
+     */
+    protected Map<MachineTranslationInfo, String> displayed = new TreeMap<>();
```

Comment:
I missed this before, but if you use `MachineTranslationInfo` as a key in a Set or Map then it needs to implement `equals` and `hashCode`.